packages/darklang/cli/local-install/install.dark

```diff
@@ -4,61 +4,118 @@ module Darklang =
       /// Helps to download the latest release of the Darklang CLI,
       /// and replace the current version with it.
       module Installation =
-        /// TODO we should really use something in Dark to get the asset name,
-        /// rather than using `basename` in the shell
-        /// (won't work for Windows, anyway.)
         let basename (url: String) : Stdlib.Result.Result<String, String> =
-          let result = Stdlib.Cli.execute $"basename {url}"
+          let basename = url |> Stdlib.String.split "/" |> Stdlib.List.last
 
-          if result.exitCode == 0L then
-            Stdlib.Result.Result.Ok(result.stdout |> Stdlib.String.trim)
-          else
-            Stdlib.Result.Result.Error "Failed to get asset name"
+          match basename with
+          | Some name -> Stdlib.Result.Result.Ok name
+          | None -> Stdlib.Result.Result.Error "Failed to get asset name"
 
 
         // TODO: delete old version of the executable, if the new version is working correctly
         let installOrUpdateLatestRelease
           (host: Stdlib.Cli.Host.Host)
           : Stdlib.Result.Result<Unit, String> =
-          (Stdlib.Cli.Unix.ensureDirectoryExists executableDir) |> Builtin.unwrap
+          match host.os with
+          | Windows ->
+            let homeDirectory =
+              (Stdlib.Cli.PowerShell.getHomeDirectory ()) |> Builtin.unwrap
 
-          let latestReleaseTag =
-            (GitHub.Releases.getLatestReleaseTag bearerToken) |> Builtin.unwrap
+            let executableDir = $"{homeDirectory}\\.darklang\\bin\\"
+            let configPath = $"{homeDirectory}\\.darklang\\config.json"
 
-          let downloadUrl = // will be something like `darklang-linux-x64.gz`
-            (GitHub.Releases.getDownloadUrl host bearerToken latestReleaseTag)
+            let ensureDirectoryExists =
+              Stdlib.Cli.PowerShell.ensureDirectoryExists executableDir
+
+            let latestReleaseTag =
+              (GitHub.Releases.getLatestReleaseTag bearerToken) |> Builtin.unwrap
+
+            let downloadUrl = // will be something like `darklang-linux-x64.gz`
+              (GitHub.Releases.getDownloadUrl host bearerToken latestReleaseTag)
+              |> Builtin.unwrap
+
+            let assetName = (basename downloadUrl) |> Builtin.unwrap
+
+            let downloadTo = $"{executableDir}{assetName}"
+
+            let downloadFileToPath =
+              Stdlib.Cli.Curl.downloadFileToPath downloadUrl downloadTo
+
+            let extractedExecutablePath =
+              if Stdlib.String.endsWith assetName ".gz" then
+                $"{executableDir}{assetName |> Stdlib.String.dropLast 3L}"
+              else
+                // TODO: error here better
+                Stdlib.Result.Result.Error "Asset is not a .gz file"
+
+            (Stdlib.Cli.PowerShell.Gzip.unzipToFile
```

Comment:
do all Windows installs come with Powershell and Gzip out of the box these days?

Reply:
Windows comes with PowerShell (Since Windows 10, PowerShell replaced Command Prompt)

As for the Gzip functionality, it comes built into the .NET Framework that PowerShell uses, so I believe it would work on older versions